src/basic/alloc-util.h

```diff
@@ -144,6 +190,14 @@ void* greedy_realloc0(void **p, size_t *allocated, size_t need, size_t size);
                 (void*)(((uintptr_t)_ptr_ + _mask_) & ~_mask_);         \
         })
 
+#define alloca_align_offset(size, align, offset)                        \
+        ({                                                              \
+                uintptr_t _ptr_;                                        \
+                size_t _mask_ = (align) - 1;                            \
+                _ptr_ = (uintptr_t)alloca((size) + _mask_) - (offset);  \
+                (void*)(((_ptr_ + _mask_) & ~_mask_) + (offset));       \
```

Comment:
not sure i fully grok this. can you add a comment here?

Reply:
hope it's a bit clearer.